src/shared/services/ServiceConfig.ts

```diff
@@ -0,0 +1,47 @@
+import axios, { AxiosInstance, AxiosResponse, AxiosRequestConfig } from 'axios'
+
+export const client: AxiosInstance = axios.create({
+  baseURL: process.env.REACT_APP_BASE_URL
+})
+
+class ServiceConfig {
+  public async get<T, B, R = AxiosResponse<T>>(
+    url: string,
+    data?: AxiosRequestConfig<B>
+  ): Promise<R> {
+    try {
+      return await client.get(url, data)
+    } catch (error) {
+      return Promise.reject(error)
+    }
+  }
+
+  public async post<T, B, R = AxiosResponse<T>>(url: string, data?: B): Promise<R> {
+    try {
+      return await client.post(url, data)
+    } catch (error) {
+      return Promise.reject(error)
+    }
+  }
+
+  public async put<T, B, R = AxiosResponse<T>>(url: string, data?: B): Promise<R> {

```

Comment:
Trocar o put para patch

Reply:
por que fazer essa alteração? put é usado para fazer alteração de um dado por completo, e patch alterar parcialmente

Reply:
sim , mas se caso o data for alterado só uma variável o patch faz melhor essa atualização